staging/src/k8s.io/client-go/tools/pager/pager.go

```diff
@@ -73,7 +73,7 @@ func New(fn ListPageFunc) *ListPager {
 // List returns a single list object, but attempts to retrieve smaller chunks from the
 // server to reduce the impact on the server. If the chunk attempt fails, it will load
 // the full list instead. The Limit field on options, if unset, will default to the page size.
-func (p *ListPager) List(ctx context.Context, options metav1.ListOptions) (runtime.Object, error) {
+func (p *ListPager) List(ctx context.Context, options metav1.ListOptions) (runtime.Object, bool, error) {
```

Comment:
is there a reasonable way to do this without changing the signature?

Reply:
I didn't come up with anything else.
The only other option that comes to my mind is to track it at the `ListPager` level and add a method to ListPager like `WasLastListPaginated`. But then the semantic is strange if you use pager for multiple requests at the same time.

I was also afraid of it, but it seems that it's not widely used, so maybe it's fine.

Reply:
it's not ideal, but our current requirement to match levels of k8s.io libraries allows it, and the work to update client-go to plumb context into client signatures means 1.18.x client-go will not be compatible with 1.17.x apiserver go signatures already